### ISSUE TITLE
write some tests for mlabns

### ISSUE THREAD

**Issue by bassosimone** [measurement-kit/measurement-kit#465]
After #463 is done, it should be nice to have some tests for mlabns. To start off, it would probably be good to have integration tests where we call directly the service.

I'd also like to have a test where we pass the body received from mlab-ns to a function and we obtain back the structure representing the response. This should allows us to assert that the library behaves even when invalid input (or unexpected!) is returned by the mlab-ns api.

**Comment by bassosimone** [comment]
We already implemented some tests.

It would be nice to also cover the following cases:

- [ ] make sure that an error is passed to callback if input leads to an invalid query because of policy, metro, and/or address family
- [ ] make sure that an error is passed to callback if `http::request()` fails (you should probably mock the implementation using templates to do that)
- [ ] make sure that an error is passed to callback if the response status is not `200`
- [ ] make sure that an error is passed to callback if:
    - [ ] json parsing fails
    - [ ] a nonexistent json field is accessed

Regarding the two latter cases, it would also be nice to distinguish between them by defining two different errors, one for proper json parsing error and the other for missing field error.

**Comment by AntonioLangiu** [comment]
Start working on this...
